### PR TITLE
MULE-19073: Exception in Catch Exception Strategy does not roll back transaction

### PR DESCRIPTION
**Pull request by pabloperalta**

### CODE REVIEW

File: core/src/main/java/org/mule/exception/TemplateMessagingExceptionStrategy.java [hunk 1]
```diff
@@ -191,12 +192,12 @@ protected MuleEvent route(MuleEvent event, Exception t)
             catch (Exception e)
             {
                 logFatal(event, e);
+                throw new MessagingException(CoreMessages.createStaticMessage("There was an error processing the catch strategy"), event, e);
```
Comment (fsgonz) [inline core/src/main/java/org/mule/exception/TemplateMessagingExceptionStrategy.java:195]: static import

File: core/src/test/java/org/mule/exception/CatchMessagingExceptionStrategyTestCase.java [hunk 2]
```diff
@@ -115,9 +117,15 @@ public void testHandleExceptionWithConfiguredMessageProcessors() throws Exceptio
     @Test
     public void testHandleExceptionWithMessageProcessorsChangingEvent() throws Exception
     {
+        when((Class<String>) mockMuleEvent.getMessage().getDataType().getType()).thenReturn(String.class);
```
Comment (fsgonz) [inline core/src/test/java/org/mule/exception/CatchMessagingExceptionStrategyTestCase.java:120]: Why is this needed?

File: modules/cxf/src/test/resources/catch-exception-strategy-flow-conf-httpn.xml [hunk 3]
```diff
@@ -83,7 +83,7 @@
             <processor-chain>
                 <script:component>
                     <script:script engine="groovy">
-                        return message.getOriginalPayload()
+                        return new String(message.getOriginalPayload())
```
Comment (fsgonz) [inline modules/cxf/src/test/resources/catch-exception-strategy-flow-conf-httpn.xml:86]: Why is this needed?

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: modules/cxf/src/test/resources/catch-exception-strategy-flow-conf-httpn.xml (hunk 3 above)
Comment (pabloperalta) [reply modules/cxf/src/test/resources/catch-exception-strategy-flow-conf-httpn.xml]: No longer needed

File: core/src/test/java/org/mule/exception/CatchMessagingExceptionStrategyTestCase.java (hunk 2 above)
Comment (pabloperalta) [reply core/src/test/java/org/mule/exception/CatchMessagingExceptionStrategyTestCase.java]: No longer needed

**Review by mbuchwald — APPROVED** [review]

**Pull request merged**